aptos-move/mvhashmap/src/types.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_aggregator::delta_change_set::DeltaOp;
+use aptos_crypto::hash::HashValue;
+use std::sync::Arc;
+
+pub type TxnIndex = u32;
+pub type Incarnation = u32;
+pub type Version = (TxnIndex, Incarnation);
+
+/// Returned as Err(..) when failed to read from the multi-version data-structure.
+#[derive(Debug, PartialEq, Eq)]
+pub enum MVDataError {
+    /// No prior entry is found.
+    NotFound,
+    /// Read resulted in an unresolved delta value.
+    Unresolved(DeltaOp),
+    /// A dependency on other transaction has been found during the read.
+    Dependency(TxnIndex),
+    /// Delta application failed, txn execution should fail.
+    DeltaApplicationFailure,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum MVCodeError {
+    /// No prior entry is found.
+    NotFound,
+    /// A dependency on other transaction has been found during the read.
+    Dependency(TxnIndex),
+}
+
+/// Returned as Ok(..) when read successfully from the multi-version data-structure.
+#[derive(Debug, PartialEq, Eq)]
+pub enum MVDataOutput<V> {
+    /// Result of resolved delta op, always u128. Unlike with `Version`, we return
+    /// actual data because u128 is cheap to copy and validation can be done correctly
+    /// on values as well (ABA is not a problem).
+    Resolved(u128),
```

Comment:
IIUC `resolved` should only be related to aggregator? Just want to double check my understanding here.

Reply:
Yes, resolved is an aggregator concept, that's why it only exists in the data-related part now